app/lib/actions/katello/repository/destroy.rb

```diff
@@ -70,11 +70,8 @@ def handle_acs_product_removal(repository)
           # Remove products from ACS's that contain no repositories which both
           # match the ACS content type and have a non-nil URL
           product = repository.product
-          repo_content_types = Set.new
-          product.repositories.each do |test_repo|
-            # we need to check id because test_repo will still contain the old, non-nil url
-            repo_content_types.add(test_repo.content_type) if (repository.id != test_repo.id) && test_repo.url.present?
-          end
+          root_ids = product.repositories.where.not(:id => repository.id).pluck(:root_id).uniq
+          repo_content_types = ::Katello::RootRepository.where(:id => root_ids).where.not(:url => [nil, '']).pluck(:content_type).uniq
```

Comment:
How about
```ruby
 repo_content_types = ::Katello::RootRepository.where(:id => product.repositories.where.not(:id => repository.id).select(:root_id)).where.not(:url => [nil, '']).distinct.pluck(:content_type)
```

Reply:
That was my original patch :) But wasnt sure about code readability here (or where to split the cmd into multiple lines).

Either patch is fine for me. Let me know if I shall update the PR per this suggestion.